libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -1281,61 +1281,49 @@ void Compass::motor_compensation_type(const uint8_t comp_type)
 
 bool Compass::consistent() const
 {
-    Vector3f primary_mag_field = get_field();
-    Vector3f primary_mag_field_norm;
+    const Vector3f primary_mag_field = get_field();
+    const Vector2f primary_mag_field_xy = Vector2f(primary_mag_field.x,primary_mag_field.y);
 
-    if (!primary_mag_field.is_zero()) {
-        primary_mag_field_norm = primary_mag_field.normalized();
-    } else {
+    if (primary_mag_field_xy.is_zero()) {
         return false;
     }
 
-    Vector2f primary_mag_field_xy = Vector2f(primary_mag_field.x,primary_mag_field.y);
-    Vector2f primary_mag_field_xy_norm;
-
-    if (!primary_mag_field_xy.is_zero()) {
-        primary_mag_field_xy_norm = primary_mag_field_xy.normalized();
-    } else {
-        return false;
-    }
+    const Vector3f primary_mag_field_norm = primary_mag_field.normalized();
+    const Vector2f primary_mag_field_xy_norm = primary_mag_field_xy.normalized();
 
     for (uint8_t i=0; i<get_count(); i++) {
-        if (use_for_yaw(i)) {
-            Vector3f mag_field = get_field(i);
-            Vector3f mag_field_norm;
-
-            if (!mag_field.is_zero()) {
-                mag_field_norm = mag_field.normalized();
-            } else {
-                return false;
-            }
+        if (!use_for_yaw(i)) {
+            // configured not-to-be-used
+            continue;
+        }
 
-            Vector2f mag_field_xy = Vector2f(mag_field.x,mag_field.y);
-            Vector2f mag_field_xy_norm;
+        const Vector3f mag_field = get_field(i);
+        const Vector2f mag_field_xy = Vector2f(mag_field.x,mag_field.y);
 
-            if (!mag_field_xy.is_zero()) {
-                mag_field_xy_norm = mag_field_xy.normalized();
-            } else {
-                return false;
-            }
+        if (mag_field_xy.is_zero()) {
+            return false;
+        }
 
-            float xyz_ang_diff = acosf(constrain_float(mag_field_norm * primary_mag_field_norm,-1.0f,1.0f));
-            float xy_ang_diff  = acosf(constrain_float(mag_field_xy_norm*primary_mag_field_xy_norm,-1.0f,1.0f));
-            float xy_len_diff  = (primary_mag_field_xy-mag_field_xy).length();
+        const Vector3f mag_field_norm = mag_field.normalized();
```

Comment:
These extra objects are adding runtime cost/weight. You should just `normalize()` the orginal field you got in place (you have to compute the xy_len_diff before you do that of course). Mutable normlization reduces flash usage by 60 bytes, and is faster at runtime.